addons/pos_restaurant/static/src/app/screens/floor_screen/floor_screen.js

```diff
@@ -610,9 +688,11 @@ export class FloorScreen extends Component {
         }
     }
     selectFloor(floor) {
+        this.saveCurrentFloorScrollPosition();
```

Comment:
it's probably more readable if you make `saveCurrentFloorScrollPosition` and `restoreFloorScrollPosition` to accept `floor` as parameter.